examples/converters.py

```diff
@@ -0,0 +1,99 @@
+# This example requires the 'members' privileged intent to use the Member converter.
+
+import typing
+
+import discord
+from discord.ext import commands
+
+intents = discord.Intents.default()
+intents.members = True
+
+bot = commands.Bot("!", intents=intents)
+
+
+@bot.command()
+async def userinfo(ctx: commands.Context, user: discord.User):
+    # In the command signature above, you can see that the `user`
+    # parameter is typehinted to `discord.User`. This means that
+    # during command invocation, the `commands` framework will
+    # attempt to convert the value passed for the `user` parameter
+    # to a `discord.User` instance.
+
+    # NOTE: typehinting acts as a converter within the `commands` framework only.
+    # In standard Python, it is use for documentation and IDE assistance purposes.
+
+    # If the conversion is successful, we will have a `discord.User` instance
+    # and can do the following:
+    await ctx.send(user.mention)
+
+@userinfo.error
+async def userinfo_error(ctx: commands.Context, error: Exception):
+    # if the conversion above fails for any reason, it will raise `commands.errors.BadArgument`
+    # so we handle this in this error handler:
+    if isinstance(error, commands.BadArgument):
+        return await ctx.send("Couldn't find that user.")
+    else:
+        # Raise the error to the global error handler
+        raise error
+
+# Manual use of converters will follow:
+@bot.command()
+async def channel_or_member(ctx: commands.Context, argument: str):
+    # NOTE: command parameters are `str` type by default, the typehint above is just for completeness' sake.
+
+    # If you use this pattern commonly, consider making a Custom Converter.
+    # See: https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#advanced-converters
+
+    # If for any reason you have an ID, but it may be of one of more
+    # items, like a `discord.TextChannel` or `discord.Member`, you can manually call the converters
+    # like the following:
+    member_converter = commands.MemberConverter()
+    try:
+        # Try and convert to a Member instance.
+        member = await member_converter.convert(ctx, argument)
+    except commands.MemberNotFound:
+        # Could not convert to a Member instance.
+        pass
+    else:
+        # We have our `member` so lets return here.
+        return await ctx.send("Member found: {}".format(member))
+
+    # Do the same for TextChannel...
+    textchannel_converter = commands.TextChannelConverter()
+    try:
+        channel = await textchannel_converter.convert(ctx, argument)
+    except commands.ChannelNotFound:
+        pass
+    else:
+        return await ctx.send("Channel found: {}".format(channel))
+
+    await ctx.send("No member or channel matching this ID was found.")
+
+@bot.command()
+async def alternative_channel_or_member(ctx: commands.Context, target: typing.Union[discord.Member, discord.TextChannel]):
```

Comment:
Personally I'm against toy names for commands like these since the point of examples is also to be ran and used. No one will really use or see a command name like this.

Fundamentally examples should still show something inherently useful (even though I know it's hard to do so). It shouldn't be hard to come up with a concept command that does use either a member or a channel and naming it appropriately.